Updating dotnet-svcutil guide for version 2.0. (#10506)
dotnet/docs: docs/core/additional-tools/dotnet-svcutil-guide.md

> ## Prerequisites

Reviewer: please update the ms.date on line 5

> # [dotnet-svcutil 2.x](#tab/dotnetsvcutil2x)

Reviewer: We'll get warnings when try to push the commits to production later on if we don't use the pre-approved values.

```suggestion
# [.NET Core 2.x](#tab/netcore2x)
```

Author: @mairaw We want to show that the usage is different depending on which version of the tool you're using, not which version of .NET you have. Changing these tab names to the .NET version doesn't really make sense. The dotnet-svcutil tool is not shipped with the .NET Core SDK and the version numbers are not related. You can use dotnet-svcutil 1.x with only .NET Core 2.x installed. 

Will these warnings be a problem? Should we not be using tabs for this and use a different approach for the older version of the tool?

Reviewer: Gotcha. Let's request them to whitelist for now and see what happens. Usually, we don't go this granular on a tool name for the tabs. They're usually large, common values like .NET Core versions, OS names, language, etc.

We try to keep our reports clean of warnings. 

Otherwise, we can come up with a different solution on how to present this but we can publish this as-is for now.

> * [.NET Core SDK](https://dotnet.microsoft.com/download) v2.1 or later versions

Reviewer: We settled on a convention to say .NET Core *version* SDK

```suggestion
* [.NET Core 2.1 SDK](https://dotnet.microsoft.com/download) or later versions
```

* Your favorite code editor

> # [dotnet-svcutil 1.x](#tab/dotnetsvcutil1x)

Reviewer: ```suggestion
# [.NET Core 1.x](#tab/netcore1x)
```

> * [.NET Core SDK](https://dotnet.microsoft.com/download) v1.0.4 or later versions

Reviewer: ```suggestion
* [.NET Core 1.0.4 SDK](https://dotnet.microsoft.com/download) or later versions
```

* Your favorite code editor

---

## Getting started

> The following example walks you through the steps required to add a web service reference to a .NET Core web project and invoke the service. You will create a .NET Core web application named _HelloSvcutil_ and will add a reference to a web service that implements the following contract:

Reviewer: ```suggestion
The following example walks you through the steps required to add a web service reference to a .NET Core web project and invoke the service. You'll create a .NET Core web application named _HelloSvcutil_ and add a reference to a web service that implements the following contract:
```

cd HelloSvcutil
```

> 

Reviewer: on line 44 can you make the following change?
the web service will be assumed to be hosted at the following address ->
let's assume the web service will be hosted at the following address

2. Create a new C# web project in that directory using the [`dotnet new`](../tools/dotnet-new.md) command as follows:

```console
dotnet new web
```

3. Install the [`dotnet-svcutil` NuGet package](https://nuget.org/packages/dotnet-svcutil) as a CLI tool:

> # [dotnet-svcutil 2.x](#tab/dotnetsvcutil2x)

Reviewer: ```suggestion

# [.NET Core 2.x](#tab/netcore2x)
```

```console
dotnet tool install --global dotnet-svcutil
```

> # [dotnet-svcutil 1.x](#tab/dotnetsvcutil1x)

Reviewer: ```suggestion
# [.NET Core 1.x](#tab/netcore1x)
```

Open the `HelloSvcutil.csproj` project file in your editor, edit the `Project` element, and add the [`dotnet-svcutil` NuGet package](https://nuget.org/packages/dotnet-svcutil) as a CLI tool reference, using the following code:

```xml
<ItemGroup>
  <DotNetCliToolReference Include="dotnet-svcutil" Version="1.0.*" />
</ItemGroup>
```

Then restore the _dotnet-svcutil_ package using the [`dotnet restore`](../tools/dotnet-restore.md) command as follows:

```console
dotnet restore
```

---

4. Run the _dotnet-svcutil_ command to generate the web service reference file as follows:

> # [dotnet-svcutil 2.x](#tab/dotnetsvcutil2x)

Reviewer: ```suggestion

# [.NET Core 2.x](#tab/netcore2x)
```

```console
dotnet-svcutil http://contoso.com/SayHello.svc
```

> # [dotnet-svcutil 1.x](#tab/dotnetsvcutil1x)

Reviewer: ```suggestion
# [.NET Core 1.x](#tab/netcore1x)
```

```console
dotnet svcutil http://contoso.com/SayHello.svc
```
---

The generated file is saved as _HelloSvcutil/ServiceReference/Reference.cs_. The _dotnet-svcutil_ tool also adds to the project the appropriate WCF packages required by the proxy code as package references.

## Using the Service Reference

1. Restore the WCF packages using the [`dotnet restore`](../tools/dotnet-restore.md) command as follows:

```console
dotnet restore
```

> 2. Find the name of the client class and operation you want to use. `Reference.cs` will contain a class that inherits from `System.ServiceModel.ClientBase`, with methods that can be used to call operations on the service. In this example we want to call the _SayHello_ service's _Hello_ operation. `ServiceReference.SayHelloClient` is the name of the client class, and has a method called `HelloAsync` that can be used to call the operation.

Reviewer: ```suggestion
2. Find the name of the client class and operation you want to use. `Reference.cs` will contain a class that inherits from `System.ServiceModel.ClientBase`, with methods that can be used to call operations on the service. In this example, you want to call the _SayHello_ service's _Hello_ operation. `ServiceReference.SayHelloClient` is the name of the client class, and has a method called `HelloAsync` that can be used to call the operation.
```

3. Open the `Startup.cs` file in your editor, and add a using statement for the service reference namespace at the top:

> ```csharp

Reviewer: ```suggestion

```csharp
```

using ServiceReference;
```

 4. Edit the `Configure` method to invoke the web service. You do this by creating an instance of the class that inherits from `ClientBase` and calling the method on the client object:

```csharp
public void Configure(IApplicationBuilder app, IHostingEnvironment env)
{
    if (env.IsDevelopment())
    {
        app.UseDeveloperExceptionPage();
    }

    app.Run(async (context) =>
    {
        var client = new SayHelloClient();
        var response = await client.HelloAsync();
        await context.Response.WriteAsync(response);
    });
}

```

5. Run the application using the [`dotnet run`](../tools/dotnet-run.md) command as follows:

```console
dotnet run
```

> 6. Navigate to the URL listed in the console (e.g. http://localhost:5000) in your web browser.

Reviewer: ```suggestion
6. Navigate to the URL listed in the console (for example, `http://localhost:5000`) in your web browser.
```

You should see the following output:
"Hello dotnet-svcutil!"

For a detailed description of the `dotnet-svcutil` tool parameters, invoke the tool passing the help parameter as follows:

> # [dotnet-svcutil 2.x](#tab/dotnetsvcutil2x)

Reviewer: ```suggestion

# [.NET Core 2.x](#tab/netcore2x)
```

```console
dotnet-svcutil --help
```

> # [dotnet-svcutil 1.x](#tab/dotnetsvcutil1x)

Reviewer: ```suggestion
# [.NET Core 1.x](#tab/netcore1x)
```
